internal/app/s3manager/bucket_view.go

```diff
@@ -37,8 +39,11 @@ func HandleBucketView(s3 S3, templates fs.FS, allowDelete bool, listRecursive bo
 		regex := regexp.MustCompile(`\/buckets\/([^\/]*)\/?(.*)`)
 		matches := regex.FindStringSubmatch(r.RequestURI)
 		bucketName := matches[1]
-		path := matches[2]
-
+		path, rqerr := url.QueryUnescape(matches[2])
+		if rqerr != nil {
+			log.Fatal(rqerr)
+			return
```

Comment:
What the reason you calling here `log.Fatal` ? 

P.S. 
[Should a Go package ever use log.Fatal and when?](https://stackoverflow.com/q/33885235)

```suggestion
			handleHTTPError(w, err)
			return
```